SAML2/XML/Signature.hs

```diff
@@ -247,8 +249,21 @@ _verifySignatureOld pks xid doc = do
   xpathbase = "/*" ++ xpathsel "Signature" ++ xpathsel "SignedInfo" ++ "//"
   xpath = xpathbase ++ ". | " ++ xpathbase ++ "@* | " ++ xpathbase ++ "namespace::*"
 
+-- | It turns out sometimes we don't get envelopped signatures, but signatures that are
+-- located outside the signed sub-tree.  Since 'verifySiganture' doesn't support this case, if
+-- you encounter it you should fall back to 'verifySignatureLegacy'.
+verifySignatureLegacy :: PublicKeys -> String -> HXT.XmlTree -> IO (Either SignatureError ())
+verifySignatureLegacy pks xid doc = catchAll $ warpResult <$> verifySignatureOld pks xid doc
+  where
+    catchAll :: IO (Either SignatureError ()) -> IO (Either SignatureError ())
+    catchAll = handle $ pure . Left . SignatureVerificationLegacyFailure . Left . (show @SomeException)
+
+    warpResult (Just True) = Right ()
```

Comment:
Not sure if this is a typo

Reply:
You mean `wrap` instead of `warp`?  No, I meant `warp` in the sense of "change shape".